drivers/counter/counter_nrfx_rtc.c

```diff
@@ -181,10 +195,12 @@ static void event_handler(nrfx_rtc_int_type_t int_type, void *p_context)
 {
 	struct device *dev = p_context;
 	struct counter_nrfx_data *data = get_dev_data(dev);
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
 
 	if (int_type == COUNTER_TOP_INT) {
 		/* Manually reset counter if top value is different than max. */
-		if (data->top != COUNTER_MAX_TOP_VALUE) {
+		if (!IS_PPI_WRAP_ENABLED() && !nrfx_config->use_ppi &&
```

Comment:
`(!IS_PPI_WRAP_ENABLED() || !nrfx_config->use_ppi) &&`
Right now `use_ppi` is unnecessarily checked when the PPI wrapping is disabled. Actually, it would be good to completely get rid of this field from this structure when the feature is not enabled.

Reply:
done. `use_ppi` used only if feature enabled.